junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/AnnotatedElementContext.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api.extension;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.util.List;
+import java.util.Optional;
+
+import org.apiguardian.api.API;
+
+@API(status = EXPERIMENTAL, since = "5.10")
```

Comment:
```suggestion
/**
 * {@code AnnotatedElementContext} encapsulates the <em>context</em> in which an
 * {@link #getAnnotatedElement() AnnotatedElement} is declared.
 *
 * <p>A {@code AnnotatedElementContext} is used to allow inspecting the field or
 * parameter the {@link org.junit.jupiter.api.io.TempDir TempDir} annotation is
 * declared on in {@link org.junit.jupiter.api.io.TempDirFactory TempDirFactory}.
 *
 * <p>This interface is not intended to be implemented by clients.
 *
 * @since 5.10
 */
@API(status = EXPERIMENTAL, since = "5.10")
```

Reply:
I wouldn't make the documentation specific to `TempDirFactory`.

Rather, I'd phrase it rather generically (since we may use it elsewhere later) and mention that it is used to access the annotated field or parameter on which a `@TempDir` annotation is declared. Or something along those lines.

Actually, maybe you could just prepend "For example, an ..." to that paragraph to achieve the same goal. 😉

Reply:
Both feedbacks have been applied with tiny rephrasing, please let me know if you would like to change anything.